src/common/ipc/unix_socket_server.cpp

```diff
@@ -0,0 +1,230 @@
+#include <libcyphal/executor.hpp>
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+#include <utility>
+
+namespace ocvsmd
+{
+namespace common
+{
+namespace ipc
+{
+namespace
+{
+
+constexpr int MaxConnections = 5;
+
+class ClientContextImpl final : public detail::ClientContext
+{
+public:
+    explicit ClientContextImpl(const int client_fd)
+        : client_fd_{client_fd}
+    {
+        CETL_DEBUG_ASSERT(client_fd != -1, "");
+
+        std::cout << "New client connection on fd=" << client_fd << ".\n";
+    }
+
+    ~ClientContextImpl() override
+    {
+        std::cout << "Closing connection on fd=" << client_fd_ << ".\n";
+
+        platform::posixSyscallError([this] {
+            //
+            return ::close(client_fd_);
+        });
+    }
+
+    ClientContextImpl(ClientContextImpl&&)                 = delete;
+    ClientContextImpl(const ClientContextImpl&)            = delete;
+    ClientContextImpl& operator=(ClientContextImpl&&)      = delete;
+    ClientContextImpl& operator=(const ClientContextImpl&) = delete;
+
+    int getFd() const
+    {
+        return client_fd_;
+    }
+
+    void setCallback(libcyphal::IExecutor::Callback::Any&& callback)
+    {
+        callback_ = std::move(callback);
+    }
+
+private:
+    const int                           client_fd_;
+    libcyphal::IExecutor::Callback::Any callback_;
+
+};  // ClientContextImpl
+
+}  // namespace
+
+UnixSocketServer::UnixSocketServer(libcyphal::IExecutor& executor, std::string socket_path)
+    : executor_{executor}
+    , socket_path_{std::move(socket_path)}
+    , server_fd_{-1}
+    , posix_executor_ext_{cetl::rtti_cast<platform::IPosixExecutorExtension*>(&executor_)}
+{
+    CETL_DEBUG_ASSERT(posix_executor_ext_ != nullptr, "");
+}
+
+UnixSocketServer::~UnixSocketServer()
+{
+    if (server_fd_ != -1)
+    {
+        platform::posixSyscallError([this] {
+            //
+            return ::close(server_fd_);
+        });
+    }
+}
+
+bool UnixSocketServer::start()
+{
+    CETL_DEBUG_ASSERT(server_fd_ == -1, "");
+
+    if (const auto err = platform::posixSyscallError([this] {
+            //
+            return server_fd_ = ::socket(AF_UNIX, SOCK_STREAM, 0);
+        }))
+    {
+        std::cerr << "Failed to create socket: " << ::strerror(err) << "\n";
+        return false;
+    }
+
+    sockaddr_un addr{};
+    addr.sun_family                        = AF_UNIX;
+    const std::string abstract_socket_path = '\0' + socket_path_;
+    CETL_DEBUG_ASSERT(abstract_socket_path.size() <= sizeof(addr.sun_path), "");
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay,hicpp-no-array-decay)
+    std::memcpy(addr.sun_path,
+                abstract_socket_path.c_str(),
+                std::min(sizeof(addr.sun_path), abstract_socket_path.size()));
```

Comment:
On BSD I'm gonna return back to usual Unix domain sockets - BSD doesn't support abstract ones.

Reply:
BTW, @thirtytwobits let me know please, do you seek that our ocvsmd could run on macOS?
For now it is not a problem (besides the above comment about abstract unix domain sockets), but I would like to know whether it worth to spend a bit extra efforts to this or not. Thnx

Reply:
It's not required for the contract work, no. It's always helpful to have for debugging.